lib/trip.rb

```diff
@@ -13,9 +15,41 @@ def initialize(input)
       @cost = input[:cost]
       @rating = input[:rating]
 
-      if @rating > 5 || @rating < 1
-        raise ArgumentError.new("Invalid rating #{@rating}")
+      unless @rating == nil
+        if @rating > 5 || @rating < 1
+          raise ArgumentError.new("Invalid rating #{@rating}")
+        end
       end
+
+      if @end_time == nil
+      elsif @start_time > @end_time
+        raise ArgumentError.new("The #{@end_time} is before the #{@start_time}")
+      end
+
+    end # ends initialize method
+
+    def finish_trip!
+      if finished?
+        raise "The trip is already finished!"
+      end
+      @end_time = Time.new
+    end # ends "finish_trip!" method
+
+    def finished?
+      return end_time != nil
+    end # ends "finished?" method
+
+    def finished_trips
+      trips.select {|trip| trip.end_time == nil}
     end
```

Comment:
I don't know that it makes sense to include a `finished_trips` instance method on `Trip`. Bringing it back to Metz, that's like asking a trip "hey, which of your trips are finished?" The trip would probably respond "I _am_ a trip, I don't _have_ trips".